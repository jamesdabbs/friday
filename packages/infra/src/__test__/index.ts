import { Testing } from 'cdk8s';

export function synth(construct: any, { name = 'construct', ...params }: Record<string, unknown> = {}) {
  const app = Testing.app();
  const chart = new construct(app, name, params);
  return Testing.synth(chart);
}

export function find(chart: any[], kind: string, name?: string) {
  return chart.find(value => {
    if (value.kind !== kind) { return false }
    if (name && value.metadata?.name !== name) { return false }

    return true
  })
}
