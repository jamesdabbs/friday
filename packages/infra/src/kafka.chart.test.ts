import { KafkaChart } from './kafka.chart';
import { synth } from './__test__';

describe(KafkaChart, () => {
  describe('with defaults', () => {
    const chart = synth(KafkaChart)

    it('matches the snapshot', () => {
      expect(chart).toMatchSnapshot()
    })
  })
});
