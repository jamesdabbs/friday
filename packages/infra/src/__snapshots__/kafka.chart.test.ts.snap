// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KafkaChart with defaults matches the snapshot 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": Object {
      "name": "kafka",
    },
  },
  Object {
    "apiVersion": "kafka.strimzi.io/v1beta2",
    "kind": "Kafka",
    "metadata": Object {
      "name": "kafka-cluster",
      "namespace": "kafka",
    },
    "spec": Object {
      "kafka": Object {
        "config": Object {
          "inter.broker.protocol.version": "2.8",
          "log.message.format.version": "2.8",
          "offsets.topic.replication.factor": 1,
          "transaction.state.log.min.isr": 1,
          "transaction.state.log.replication.factor": 1,
        },
        "listeners": Array [
          Object {
            "name": "plain",
            "port": 9092,
            "tls": false,
            "type": "internal",
          },
          Object {
            "name": "tls",
            "port": 9093,
            "tls": true,
            "type": "internal",
          },
        ],
        "replicas": 1,
        "storage": Object {
          "type": "jbod",
          "volumes": Array [
            Object {
              "deleteClaim": false,
              "id": 0,
              "size": "1Gi",
              "type": "persistent-claim",
            },
          ],
        },
        "version": "2.8.0",
      },
      "zookeeper": Object {
        "replicas": 1,
        "storage": Object {
          "deleteClaim": false,
          "size": "1Gi",
          "type": "persistent-claim",
        },
      },
    },
  },
]
`;
